doc/OnlineDocs/contributed_packages/trustregion.rst

```diff
@@ -0,0 +1,184 @@
+Pyomo External Functions.
+
+This work was conducted as part of the Institute for the Design of Advanced
+Energy Systems (`IDAES <https://idaes.org>`_) with support through the
+Simulation-Based Engineering, Crosscutting Research Program within the U.S.
+Department of Energy’s Office of Fossil Energy and Carbon Management.
+
+.. _Eason & Biegler, 2018: https://doi.org/10.1002/aic.16364
+.. _Yoshio & Biegler, 2021: https://doi.org/10.1016/j.compchemeng.2021.107447
+
+Methodology Overview
+---------------------
+
+The formulation of the original hybrid problem is:
+
+.. math::
+    \begin{align*}
+    \displaystyle \min_{} & ~~ f\left(z, w, d\left(w\right)\right) & \\
+    \displaystyle \text{s.t.} \quad \: & ~~ h\left(z, w, d\left(w\right)\right) = 0 \\
+    \displaystyle & ~~ g\left(z, w, d\left(w\right)\right) \leq 0
+    \end{align*}
+
+where:
+
+* :math:`w \in \mathbb{R}^m` are the inputs to the external functions
+* :math:`z \in \mathbb{R}^n` are the remaining decision variables (i.e., degrees of freedom)
+* :math:`d(w) : \mathbb{R}^m \to \mathbb{R}^p` are the outputs of the external functions as a function of :math:`w`
+* :math:`f`, `h`, `g`, `d` are all assumed to be twice continuously differentiable
+
+This formulation is reworked to separate all external function information as
+follows to enable the usage of the trust region method:
+
+.. math::
+    \begin{align*}
+    \displaystyle \min_{x} & ~~ f\left(x\right) & \\
+    \displaystyle \text{s.t.} \quad \: & ~~ h\left(x\right) = 0 \\
+    \displaystyle & ~~ g\left(x\right) \leq 0 \\
+    \displaystyle & ~~ y = d\left(w\right)
+    \end{align*}
+
+where:
+
+* :math:`y \in \mathbb{R}^p` are the outputs of the external functions
+* :math:`x^T = [w^T, y^T, z^T]` is a set of all inputs and outputs
+
+Using this formulation and a user-supplied low-fidelity/ideal model basis function
+:math:`b\left(w\right)`, the algorithm iteratively solves subproblems using
+the surrogate model:
+
+.. math::
+    \begin{align*}
+    r_k\left(w\right) = b\left(w\right) + \left( d\left(w_k\right) - b\left(w_k\right) \right) + \left( \nabla d\left(w_k\right) - \nabla b\left(w_k\right) \right)^T \left( w - w_k \right)
+    \end{align*}
+
+This acts similarly to Newton's method in that small, incremental steps are taken
+towards an optimal solution. At each iteration, the current solution of the
+subproblem is compared to the previous solution to ensure that
+the iteration has moved in a direction towards an optimal solution. If not true,
+the step is rejected. If true, the step is accepted and the surrogate
+model is updated for the next iteration.
+
+When using TRF, please consider citing the above papers.
+
+TRF Inputs
+-----------
+
+The required inputs to the TRF ``solve`` method are the following:
+
+* The optimization model
+* List of degree of freedom variables within the model
+
+The optional input to the TRF ``solve`` method is the following:
```

Comment:
There is some neat Sphinx formatting you can use to turn ``solve`` into a link to the API docs automatically. I think it's something like: 
:py:meth:`solve <pyomo.contrib.trustregion.TRF.TrustRegionSolver.solve>`